kernel/src/hil/digest.rs

```diff
@@ -56,11 +93,47 @@ pub trait Digest<'a, const L: usize> {
     /// error with error code ENOSUPPORT.
     fn run(&'a self, digest: &'static mut [u8; L])
         -> Result<(), (ErrorCode, &'static mut [u8; L])>;
+}
 
-    /// Clear the keys and any other sensitive data.
-    /// This won't clear the buffers provided to this API, that is up to the
-    /// user to clear.
-    fn clear_data(&self);
+/// Computes a digest (cryptographic hash) over data
+///
+/// 'L' is the length of the 'u8' array to store the digest output.
+pub trait DigestVerify<'a, const L: usize> {
+    /// Compare the specified digest in the `compare` buffer to the calculated
+    /// digest. This function is similar to `run()` and should be used instead
+    /// of `run()` if the caller doesn't need to know the output, just if it
+    /// matches a known value.
+    ///
+    /// For example:
+    /// ```rust,ignore
+    ///     // Compute a digest on data
+    ///     add_data(...);
+    ///     add_data(...);
```

Comment:
Can you call `add_data` twice synchronously?

Reply:
No, you have to wait for the callback. This is just pseudo code, I'll add a comment